compiler/rustc_trait_selection/src/error_reporting/infer/note_and_explain.rs

```diff
@@ -313,11 +313,9 @@ impl<T> Trait<T> for X {
                     (ty::Dynamic(t, _, ty::DynKind::Dyn), _)
                         if let Some(def_id) = t.principal_def_id() =>
                     {
-                        let mut impl_def_ids = vec![];
-                        tcx.for_each_relevant_impl(def_id, values.found, |did| {
-                            impl_def_ids.push(did)
-                        });
-                        if let [_] = &impl_def_ids[..] {
+                        let has_non_blanket_impl =
```

Comment:
You might instead want to use a `DeepRejectCtxt` to filter out impls that obviously don't apply, rather than only considering non-blanket-impls.

Reply:
Sorry, actually let me not phrase that as a suggestion, haha. Could you please try using a `DeepRejectCtxt` here? :D